src/validations/rules/Writing_FedRAMP_Validations.md

```diff
@@ -0,0 +1,60 @@
+
+-[Guide to OSCAL-based FedRAMP Security Assessment Reports (SAR)](./Guide_to_OSCAL-based_FedRAMP_Security_Assessment_Reports_(SAR).pdf)
+
+-[Guide to OSCAL-based FedRAMP Plan of Action and Milestones (POA&M)](./Guide_to_OSCAL-based_FedRAMP_Plan_of_Action_and_Milestones_(POAM).pdf)
+
+Each describes how FedRAMP expects each OSCAL document type is to be presented in an authorization package.
+
+# FedRAMP validation constraints
+
+FedRAMP validation constraints extend the NIST OSCAL XML Schema constraints. The XML Schema constraints are essentially syntactic; Schematron is used to augment the former with semantic constraints which cannot be expressed using XML Schema.
+
+Schematron constraints have thus far been for system security plan documents. They are found in a Schematron document `ssp.sch`.
+
+A companion document `fedramp_values.xml` defines sets of allowed values for various FedRAMP OSCAL XML  elements and their attributes.
+
+An additional Schmeatron document `sch.sch` defines constraints for the FedRAMP Schematron constraints themselves, guiding a FedRAMP constraint author to augment the fundamental Schematron with FedRAMP-specific constructs. These augmentations provide
+
+- References to associated unit tests
+- References to documentation related to the constraint
+- References to associated diagnostic messages
+- Forward and reverse linkages between Schematron rules (`<rule>` elements), assertions (`<assert>` elements, and `<diagnostic>` elements.
+
+# Examples of Schematron constraints
+
+## Example 1
+
+Consider this [code block](https://github.com/18F/fedramp-automation/blob/develop/src/validations/rules/ssp.sch#L2176-L2213).
+
+The `<sch:rule>` element sets the locus within the document to be used as a relative reference for the subordinate assertions (`<sch:assert>` elements). In this case, the locus is `oscal:system-characteristics`: an XPath statement identifying the context element using a (namespace-)qualified element name.
+
+The first assertion `id="has-system-id"`has a test (an XPath statement in the `test` attribute stating `oscal:system-id[@identifier-type eq 'https://fedramp.gov']` which asserts "the `oscal:system-characteristics` element has a child element `oscal:system-id` which must have an `@identifier-type eq 'https://fedramp.gov'` with a specific value. The natural language prose equivalent "A FedRAMP SSP must have a FedRAMP system identifier." resides within the `<sch:assert>` element.
+
+The next two assertions are similar, each stating a particular child element must be present.
+
+A reference to the `fedramp_values.xml` document sets a variable to the desired [`<value-set>`](https://github.com/18F/fedramp-automation/blob/develop/dist/content/resources/xml/fedramp_values.xml#L83-L92) allowed values. The following assertion (`id="has-fedramp-authorization-type"`) tests that a child `<oscal:prop>` element has the desired attributes.
```

Comment:
This is a good start, but I am not sure it explains to me how to modify, add, or remove a rule in a way sufficiently explanatory to someone who does not have sufficiently deep experience in XML technologies, XDM (XML Document Model), Schematron, or its underlying technologies.

I think we should meet tomorrow and discuss further.

Reply:
Added several examples in https://github.com/18F/fedramp-automation/commit/bd71afaefcda11214aac1b0a6fd84d7f3e9a7762